wgpu-hal/src/vulkan/command.rs

```diff
@@ -52,6 +51,26 @@ impl super::CommandEncoder {
             }
         }
     }
+
+    fn make_framebuffer(
+        &mut self,
+        key: super::FramebufferKey,
+    ) -> Result<vk::Framebuffer, crate::DeviceError> {
+        Ok(match self.framebuffers.entry(key) {
+            Entry::Occupied(e) => *e.get(),
+            Entry::Vacant(e) => {
+                let vk_info = vk::FramebufferCreateInfo::default()
+                    .render_pass(e.key().raw_pass)
+                    .width(e.key().extent.width)
+                    .height(e.key().extent.height)
+                    .layers(e.key().extent.depth_or_array_layers)
+                    .attachments(&e.key().attachments);
```

Comment:
question: Would destructuring be helpful here for refactor-proof code?

Reply:
That's a good point since all the fields of the key should be used, if not they should be removed; will update.